olp-cpp-sdk-core/include/olp/core/network2/NetworkRequest.h

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (C) 2019 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+#pragma once
+
+#include <cstdint>
+#include <memory>
+#include <string>
+#include <unordered_map>
+#include <vector>
+
+#include <olp/core/CoreApi.h>
+
+namespace olp {
+namespace network2 {
+
+class CORE_API NetworkRequest {
+ public:
+
+  enum class HttpVerb { GET = 0, POST = 1, HEAD = 2, PUT = 3, DEL = 4, PATCH = 5 };
+
+  NetworkRequest(const std::string& url);
+
+  const std::unordered_map<std::string, std::string>& GetHeaders() const;
+  const std::string& GetUrl() const;
+  HttpVerb GetVerb() const;
+  std::shared_ptr<const std::vector<std::uint8_t>> GetBody() const;
+
+  NetworkRequest& SetHeader(std::string name, std::string value);
```

Comment:
Same here as in Proxy

```suggestion
  NetworkRequest& WithHeader(std::string name, std::string value);
  NetworkRequest& WithHeaders(std::vector<std::pair<std::string, std::string>> headers);
```